tools/nut-scanner/nut-scanner.c

```diff
@@ -305,6 +316,15 @@ int main(int argc, char *argv[])
 					timeout = DEFAULT_TIMEOUT*1000*1000;
 				}
 				break;
+                        case 'T' :
+#ifdef HAVE_PTHREAD
+                                thread_number = atol(optarg);
+                                if( thread_number <= 0 ) {
```

Comment:
Space in parentheses is different from neighboring code and (IIRC) the NUT coding style guide.

Reply:
Per my reading of docs and discussions like https://stackoverflow.com/questions/9369873/sem-init-what-is-the-value-parameter-for, a `sem_init()` using `value==0` *might* never pass the semaphore: the `sem_wait` will decrement it into negative, and block waiting until nobody progresses to `sem_post()` to release the lock. Please take another look at it, or require that the value is positive (at least `1`) right away?